mvaTraining/common.py

```diff
@@ -0,0 +1,231 @@
+
+    def load_backgrounds(self, add_mass_column=False):
+
+        if add_mass_column and not self.resonant_mass_probabilities:
+            raise Exception("You need to first load the resonant signal before the background")
+
+        datasets = []
+        weights = []
+
+        print("Loading backgrounds...")
+
+        for background in backgrounds:
+            f = get_file_from_glob(os.path.join(INPUT_FOLDER, background['input']))
+            dataset, weight = tree_to_numpy(f, self.variables, self.weight_expression, self.selection, reweight_to_cross_section=True)
+            weights.append(weight)
+
+            if add_mass_column:
+                rs = np.random.RandomState(42)
+                mass_col = np.array(rs.choice(self.resonant_mass_probabilities[0], len(dataset), p=self.resonant_mass_probabilities[1]), dtype='float')
+                dataset = np.c_[dataset, mass_col]
+
+            datasets.append(dataset)
+
+        self.background_dataset = np.concatenate(datasets)
+        self.background_weights = np.concatenate(weights)
+
+        print("Done. Number of background events: %d ; Sum of weights: %.4f" % (len(self.background_dataset), np.sum(self.background_weights)))
+
+
+    def split(self, reweight_background_training_sample=True):
+        """
+        Split datasets into a training and testing samples
+
+        Parameter:
+            reweight_background_training_sample: If true, the background training sample is reweighted so that the sum of weights of signal and background are the same
+        """
+
+        self.train_signal_dataset, self.test_signal_dataset, self.train_signal_weights, self.test_signal_weights = train_test_split(self.signal_dataset, self.signal_weights, test_size=0.5, random_state=42)
+        self.train_background_dataset, self.test_background_dataset, self.train_background_weights, self.test_background_weights = train_test_split(self.background_dataset, self.background_weights, test_size=0.5, random_state=42)
+
+        if reweight_background_training_sample:
+            sumw_train_signal = np.sum(self.train_signal_weights)
+            sumw_train_background = np.sum(self.train_background_weights)
+
+            ratio = sumw_train_signal / sumw_train_background
+            self.train_background_weights *= ratio
+            print("Background training sample reweighted so that sum of event weights for signal and background match. Sum of event weight = %.4f" % (np.sum(self.train_signal_weights)))
+
+        # Create merged training and testing dataset, with targets
+        self.training_dataset = np.concatenate([self.train_signal_dataset, self.train_background_dataset])
+        self.training_weights = np.concatenate([self.train_signal_weights, self.train_background_weights])
+        self.testing_dataset = np.concatenate([self.train_signal_dataset, self.train_background_dataset])
+        self.testing_weights = np.concatenate([self.train_signal_weights, self.train_background_weights])
```

Comment:
Indeed... I have one already fixed locally, but not eveything. Thanks a lot for the review :) I'll do a PR in the near future with the non-resonant network too :)